src/compiler/transformers/jsx.ts

```diff
@@ -40,17 +40,25 @@ namespace ts {
         }
 
         function getImplicitImportForName(name: string) {
-            const existing = currentFileState.utilizedImplicitRuntimeImports?.get(name);
+            const importSource = name === "createElement"

```

Comment:
this is a special case for the `createElement` within this "generic" util, but in the name of KISS I think it's perfectly OK here. The transform is tied to the semantics defined by React anyhow, this won't change often and there is near-zero chance that `createElement` will get reintroduced in the future in some other entrypoint in JSX runtimes